feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/validators/ValidatorProvider.kt

```diff
@@ -91,7 +91,10 @@ class ValidatorProvider(
 
         val slashes = stakingRepository.getSlashes(chainId, accountIdBridged)
 
-        val novaValidatorIds = knownNovaValidators.getValidatorIds(chainId).toSet()
+        val novaValidatorIds = validatorsPreferencesSource.getRecommendedValidatorIds(chainId)
```

Comment:
I don't think we need to apply exclusion logic when we are getting a validator. I think the cause we are using preferred one is due to the validator might not be elected

Reply:
Agree, thank you!